index.d.ts

```diff
@@ -0,0 +1,490 @@
+    | 'US'
+    | 'UY'
+    | 'UZ'
+    | 'VU'
+    | 'VA'
+    | 'VE'
+    | 'VN'
+    | 'WF'
+    | 'EH'
+    | 'YE'
+    | 'ZM'
+    | 'ZW'
+    | 'AX';
+
+// source: https://developers.google.com/maps/faq#languagesupport
+type Vue2GeocoderLanguageCode =
+    | 'ja'
+    | 'sq'
+    | 'kn'
+    | 'am'
+    | 'kk'
+    | 'ar'
+    | 'km'
+    | 'hy'
+    | 'ko'
+    | 'az'
+    | 'ky'
+    | 'eu'
+    | 'lo'
+    | 'be'
+    | 'lv'
+    | 'bn'
+    | 'lt'
+    | 'bs'
+    | 'mk'
+    | 'bg'
+    | 'ms'
+    | 'my'
+    | 'ml'
+    | 'ca'
+    | 'mr'
+    | 'zh'
+    | 'mn'
+    | 'zh-CN'
+    | 'ne'
+    | 'zh-HK'
+    | 'no'
+    | 'zh-TW'
+    | 'pl'
+    | 'hr'
+    | 'pt'
+    | 'cs'
+    | 'pt-BR'
+    | 'da'
+    | 'pt-PT'
+    | 'nl'
+    | 'pa'
+    | 'en'
+    | 'ro'
+    | 'en-AU'
+    | 'ru'
+    | 'en-GB'
+    | 'sr'
+    | 'et'
+    | 'si'
+    | 'fa'
+    | 'sk'
+    | 'fi'
+    | 'sl'
+    | 'fil'
+    | 'es'
+    | 'fr'
+    | 'es-419'
+    | 'fr-CA'
+    | 'sw'
+    | 'gl'
+    | 'sv'
+    | 'ka'
+    | 'ta'
+    | 'de'
+    | 'te'
+    | 'el'
+    | 'th'
+    | 'gu'
+    | 'tr'
+    | 'iw'
+    | 'uk'
+    | 'hi'
+    | 'ur'
+    | 'hu'
+    | 'uz'
+    | 'is'
+    | 'vi'
+    | 'id'
+    | 'zu'
+    | 'it';
+
+// source: https://developers.google.com/maps/documentation/geocoding/overview#Types
+type AddressComponentType =
+    | 'street_address'
+    | 'route'
+    | 'intersection'
+    | 'political'
+    | 'country'
+    | 'administrative_area_level_1'
+    | 'administrative_area_level_2'
+    | 'administrative_area_level_3'
+    | 'administrative_area_level_4'
+    | 'administrative_area_level_5'
+    | 'colloquial_area'
+    | 'locality'
+    | 'sublocality'
+    | 'neighborhood'
+    | 'premise'
+    | 'subpremise'
+    | 'plus_code'
+    | 'postal_code'
+    | 'natural_feature'
+    | 'airport'
+    | 'park'
+    | 'point_of_interest'
+    | 'floor'
+    | 'establishment'
+    | 'landmark'
+    | 'point_of_interest'
+    | 'parking'
+    | 'post_box'
+    | 'postal_town'
+    | 'room'
+    | 'street_number'
+    | 'bus_station';
+
+// source: https://developers.google.com/maps/documentation/geocoding/overview#results
+type LocationType =
+    | 'ROOFTOP'
+    | 'RANGE_INTERPOLATED'
+    | 'GEOMETRIC_CENTER'
+    | 'APPROXIMATE';
+
+interface AddressComponent {
+    long_name: string;
+    short_name: string;
+    types: AddressComponentType[];
+}
+
+interface Bound {
+    lat: number;
+    lng: number;
+}
+
+interface Bounds {
+    northeast: Bound;
+    southwest: Bound;
+}
+
+interface Location {
+    lat: number;
+    lng: number;
+}
+
+interface Viewport {
+    northeast: Bound;
+    southwest: Bound;
+}
+
+interface Geometry {
+    bounds: Bounds;
+    location: Location;
+    location_type: LocationType;
+    viewport: Viewport;
+}
+
+export interface Result {
+    address_components: AddressComponent[];
+    formatted_address: string;
+    geometry: Geometry;
+    place_id: string;
+    types: string[];
```

Comment:
Once we update the `AddressComponentType` type to `AddressType` it makes sense to do this

```suggestion
    types: AddressType[];
```